components/dispatcher/internal/client/client.go

```diff
@@ -37,10 +37,9 @@ type retriableCloudEventClient struct {
 
 var _ RetriableClient = &retriableCloudEventClient{}
 
-func NewRetriableCloudEventClient(uri string, exponentialBackoff bool, initialRetryInterval int64, maxRetryTime int64) retriableCloudEventClient {
+func NewRetriableCloudEventClient(exponentialBackoff bool, initialRetryInterval int64, maxRetryTime int64) retriableCloudEventClient {
 	transport, err := cloudevents.NewHTTPTransport(
-		cloudevents.WithTarget(uri),
-		cloudevents.WithEncoding(cloudevents.HTTPBinaryV03),
+		cloudevents.WithBinaryEncoding(),
 	)
 	transport.Client = httpClient
 
```

Comment:
transport could be nil here - should move to after error handling?

Reply:
Good call.